src/orange/surf/LocalSurfaceVisitor.hh

```diff
@@ -136,13 +131,13 @@ LocalSurfaceVisitor::operator()(F&& func, LocalSurfaceId id)
 template<class T>
 CELER_FUNCTION T LocalSurfaceVisitor::make_surface(LocalSurfaceId id) const
 {
-    OpaqueId<real_type> offset
+    using ItemIdT = typename decltype(params_.reals)::ItemIdT;
+    ItemIdT offset
         = this->get_item(params_.real_ids, surfaces_.data_offsets, id);
     constexpr size_type size{T::StorageSpan::extent};
     CELER_ASSERT(offset + size <= params_.reals.size());
 
-    real_type const* data = this->get_ptr(params_.reals, offset);
-    return T{Span<real_type const, size>{data, size}};
+    return T{params_.reals[{offset, offset + size}]};
```

Comment:
Have you tested this is using LDGIterator and works on CUDA? Currently the surface constructors are written as
```c++
CELER_FUNCTION Plane::Plane(StorageSpan data)
    : normal_{data[0], data[1], data[2]}, d_{data[3]}
{}
```
where `StorageSpan` is a `Span<real_type const>`. We might need to template these constructors (`template<T>` and `Span<T>`? or `T const` and we need to change where the const in LdgValue belongs)?

Reply:
yes, I'm building with CUDA, and tests pass locally. You can copy construct a `Span<T const>` from a `LdgSpan<T>` and the other way around. I tried to change the `StorageSpan` of [surfaces](https://github.com/celeritas-project/celeritas/pull/1000#issuecomment-1786195481) to `LdgSpan`. I haven't investigated why it breaks. Are these getting written to and read during a kernel?

Reply:
Read only, and it should always be from global memory too... but maybe there's some place it's constructing from a different memory space? As long as both span and ldgspan are available it should be ok...

Reply:
I guess I should clarify: on device in the main code they're always LDG, but they might not be using immutable device memory in some of the tests.